File: app/src/lib/request/continent.ts
// Copyright (C) Call of Nil contrireadonlyors
// SPDX-License-Identifier: AGPL-3.0-only

export interface GetContinentSizeRequest {
  readonly world: WorldId;
}

export interface GetPublicFieldRequest {
  readonly world: WorldId;
  readonly coord: Coord;
}

export interface GetPublicFieldsRequest {
  readonly world: WorldId;
  readonly coords: readonly Coord[];
}

File: crates/nil-payload/src/continent.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::continent::Coord;
use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetContinentSizeRequest {
  pub world: WorldId,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPublicFieldRequest {
  pub world: WorldId,
  pub coord: Coord,
}
#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GetPublicFieldsRequest {
  pub world: WorldId,
  pub coords: Vec<Coord>,
}
